### DISCUSSION TITLE
[Question] How Set canvas size ?

### DISCUSSION THREAD

**Discussion by santoshgistto** [hm21/pro_image_editor#97] [Q&A] [answered]
### Placeholder

How Set canvas size ?

**Comment by hm21** [comment]
Can you describe more in detail what you mean?

**Comment by adamkoch** [comment]
Actually I had a similar question, not sure if the same as what @santoshgistto is asking, but in my case, I want to input a fixed ratio image and then only have edits allowed on the image and not outside the image. So canvas size == image size in my case.

Here's an example of a 1:1 ratio image being loaded and what I mean:
![Simulator Screenshot - iPhone 15 Pro - 2024-06-08 at 18 41 29](https://github.com/hm21/pro_image_editor/assets/869261/9e3ab80d-9acf-4e20-8af0-7b4aab6b74b7)
![Simulator Screenshot - iPhone 15 Pro - 2024-06-08 at 18 42 15](https://github.com/hm21/pro_image_editor/assets/869261/88bdd460-56e3-49c3-98f1-9a0ae7003df6)


So ideally I want to have 1:1 image in, and edits only allowed within the image and then save and output 1:1 image.

**Comment by hm21** [comment]
It's currently not possible to do this directly in the editor, but I'm adding it to my roadmap. 
However, if you need this feature right now, you can do it yourself by using the [image](https://pub.dev/packages/image) package. An example of how to crop the image to a specific size can be seen [here](https://github.com/hm21/pro_image_editor/blob/82abffcdf8226768029882282eb6f5a4288f2a8d/lib/utils/content_recorder.dart/utils/convert_raw_image.dart#L113).
And don't forget to set in the generation configs that it capture the full editor area, like below.
```dart
 ImageGeneratioConfigs(
      generateOnlyDrawingBounds: false,
    )
```

**Comment by adamkoch** [comment]
Got it, thanks. Right now I crop the image myself already and pass it into pro_image_editor so it comes out the right size. `generateOnlyDrawingBounds=false` is a great tip for now though 👍

**Comment by santoshgistto** [comment]
Thank you

**Comment by adamkoch** [comment]
Thanks for all the fixes/updates! So is the aspect ratio thing I mentioned addressed in 3.0.5? If so, I couldn't see how to enable it?

**Comment by hm21** [comment]
@adamkoch  Not yet, I just moved it to the roadmap so I closed the question.

**Comment by hm21** [comment, marked as answer]
@adamkoch I just added the option to capture only the background image area, which we can enable in the `imageGenerationConfigs` with the `captureOnlyBackgroundImageArea` flag. This flag is now enabled by default. If a layer is outside the background image, it will show the user a bit hidden because I overlay the background color with opacity on the layer. If you want to hide it completely, you can set the opacity from the option `outsideCaptureAreaLayerOpacity` to 1 in the configs of the `imageEditorTheme` like below:

```dart
 configs: ProImageEditorConfigs(
      imageEditorTheme: const ImageEditorTheme(
                outsideCaptureAreaLayerOpacity: 1,
   ),
)
```

I didn't release this version as a stable release, but as a prerelease `3.0.7-dev.1` so that you can test it first. After you test it and the case that everything works fine by you, I will release it as a stable release.

**Comment by adamkoch** [comment]
Thank you @hm21. I tried and it works really well. With `outsideCaptureAreaLayerOpacity: 1,` is what I was after.

I did a weird glitch in landscape mode on iphone simulator - the drawing overlay seems to show in the safearea zone even though there is a black color set for the Scaffold. It may just be a glitch on the simulator. I don't have a real device to test right now. But I think it's a fairly minor thing.

![Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 22 35 53](https://github.com/hm21/pro_image_editor/assets/869261/cefe9bfc-c27e-452b-bfe6-64f9c8b24d93)

**Comment by adamkoch** [comment]
Oh also, I realized it's not a great idea to set `outsideCaptureAreaLayerOpacity: 1` because deleting items means dragging the item over a completely blacked out area which is a bit confusing. So probably still better to keep it at some value <1.
